### PR TITLE
Reorganize test structure

### PR DESCRIPTION
**Pull request by BjarkeMeier**
## Description

- Clean up of organization of tests in projects and folders. No code or functionality has been changed.
- Unused code has been removed.
- `[IntegrationTest]` attribute has been added to all integration tests. This should now be seen as the guideline.
- We (@prtandrup and @BjarkeMeier) decided to not introduce domain test project or attribute for now

The modified structure can be seen on this image:

image

## References

* #691

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeRepositoryTests.cs [hunk 1]
```diff
@@ -32,12 +30,12 @@
 using Charge = GreenEnergyHub.Charges.Domain.Charges.Charge;
 using MarketParticipant = GreenEnergyHub.Charges.Infrastructure.Context.Model.MarketParticipant;
 
-namespace GreenEnergyHub.Charges.Tests.Infrastructure.Repositories
+namespace GreenEnergyHub.Charges.IntegrationTests.IntegrationTests.Repositories
 {
     /// <summary>
     /// Tests <see cref="ChargeRepository"/> using a database created with squadron.
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeRepositoryTests.cs:36]:
```suggestion
    /// Tests <see cref="ChargeRepository"/> using a SQL database.
```

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/DefaultChargeLinkRepositoryTests.cs [hunk 2]
```diff
@@ -21,12 +21,12 @@
 using Xunit;
 using Xunit.Categories;
 
-namespace GreenEnergyHub.Charges.Tests.Infrastructure.Repositories
+namespace GreenEnergyHub.Charges.IntegrationTests.IntegrationTests.Repositories
 {
     /// <summary>
     /// Tests <see cref="DefaultChargeLinkRepository"/> using an SQLite in-memory database.
```
Comment (HenrikSommer) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/DefaultChargeLinkRepositoryTests.cs:27]:
```suggestion
    /// Tests <see cref="DefaultChargeLinkRepository"/> using a SQL database.
```

**Review by HenrikSommer — APPROVED** [review]
Approve with a couple of suggestions, that could be relevant for more repository tests.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeRepositoryTests.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.IntegrationTests/IntegrationTests/Repositories/ChargeRepositoryTests.cs:36]: Thank you for your suggestions. I, however, prefer not to integrate them in order to keep the PR solely focused on folder structure and test attributes and nothing else.

**Pull request merged**